tests/utilities/test_tools.py

```diff
@@ -69,6 +73,66 @@ def test_asciistr(self):
             asciistr(53)
 
 
+class TestCSVWriter:
+    def test_csv_writer(self, tmp_path):
```

Comment:
Something like the below fed through avoids duplicate code 
```suggestion
    @pytest.parametrize(('ext', 'comment), [('.csv', '#'), ('.txt','!')]) 
    def test_csv_writer(self, ext, comment, tmp_path):
```